### DISCUSSION TITLE
Remix PWA v3.0

### DISCUSSION THREAD

**Discussion by ShafSpecs** [ShafSpecs/remix-pwa#120] [Announcements]
After weeks of building, iteration, debugging and testing, we are finally please to announce Remix PWA v3.0 to you! The third major release of `remix-pwa` ships with a lot more changes, updates, fixes and even additions you didn't see coming. A huge thanks to all of you for your contributions, issues, PRs, ideas, RFCs and discussions. This was one huge teamwork and a big W for PWA in the Remix community 💿! 

## ✨ Features

It's more than just new features but rather a whole ecosystem shipped this time around 😁.

 - Remix PWA just got it's own compiler! Remix PWA v3.0 ships with a brand new compiler for building service workers and a lot more. Hook your configs into `remix.config.js` and build away! 
 - A brand new CLI engine. `remix-pwa` utilizes a new CLI engine for it's executable that also hooks into the compiler for a blazingly, fast setup experience https://github.com/remix-pwa/monorepo/pull/6
 - New worker route APIs. Say hello to `workerLoader` and `workerAction`, handler functions in your route file that run exclusively in the worker thread.
 - A whole new set of packages shipped including `@remix-pwa/client` and `@remix-pwa/sync`
 - `@remix-pwa/sw` got a major bump! Still the same old package you knew with a few additions and removals.
 - `@remix-pwa/strategy` just landed. The new breakaway from the `sw` package, introducing a new approach to handle caching strategies in `remix-pwa` https://github.com/remix-pwa/monorepo/pull/35.
 - A new strategy joined the troupe! `StaleWhileRevalidate` is the latest addition to the famous four you knew from the previous versions https://github.com/remix-pwa/monorepo/commit/acf8f9e6b59d15d0d4097a93c60388126607c317
 - New component: [`LiveReload`](https://remix-pwa.run/docs/live-reload). Which is simply a Remix's `LiveReload` with HMR support for service workers.
 - New PWA templates.
 - Native Background Sync support. Be assured that a request would happen even after closing your app 🔄
 - `@remix-pwa/cache` has landed 🚀! Introducing `RemixCache`: a thin wrapper around the browser `Cache` API with support for expiration and sizing. Also ships support for `cachified` out of the box 📦  
 - New Service Worker exports: `errorHandler`, `defaultFetchHandler` and `getLoadContext` to help you take service workers to the next level. Inspired by none other than Remix!
 - A brand new site 🌐! Take a peek at the brand new [Remix PWA site](https://remix-pwa.run).
 - Many more...! We can't list them all, take a look at the site if you will.

## 🐛 Bug Fixes

- Completely deleted the old `remix-pwa` executable, we shan't be hearing from them in a long time.
- A lot of minor fixes and tweaks but since they're all v3.0-related, I don't think they count.

## 💅 Enhancements

- Separated `sw` and `strategy`. Now, you can utilise a service worker easily and cache at your whim without bordering on caching.
- Remix PWA is now a PWA framework as opposed to a caching solution.

## 😎 Contributors

- @ShafSpecs 
- @Sarabadu
- @lfantone

Thanks to all the amazing contributors that made this whole thing possible ❤️!

![Remix PWA v3.0](https://ucarecdn.com/43135828-6822-4549-a0b7-23219e7353d1/-/preview/1200x630/-/quality/smart_retina/-/format/webp/)

<hr /><em>This discussion was created from the release <a href='https://github.com/remix-pwa/remix-pwa/releases/tag/v3.0.0'>Remix PWA v3.0</a>.</em>

**Comment by fernandowinfield** [comment]
Hi there, thanks for this new release! I'm wondering if these are the full notes or if there are additional notes/documentation coming for the new stuff in v3.0? The link in the readme (where it says "Remix PWA v3.0 is finally out! Check out the full release notes here") takes me to a 404 page

**Reply by ShafSpecs** [reply]
Yes, they are. I made a mistake with the release notes, I linked `.../v3.0` instead of `.../v3.0.0`

**Comment by rubinovitz** [comment]
Congrats! Did this remove push notification support? I see it's no longer a package listed in the documentation.

**Reply by ShafSpecs** [reply]
Nope, push is still available. Yet a bit broken (I haven't updated `push` package.json yet), but you can still make it work via a `--force` install.

As for it's presence in the docs, I am working on something new for it, so in the meantime, I am not porting the docs over 👀

**Reply by rubinovitz** [reply]
@ShafSpecs got it, thank you!!! 🚀

**Comment by rista404** [comment]
Hi, is this ready to be migrated to? Is there a migration guide we need to know about or just a simple version change?
I still can't see the latest version on npm.

Thanks!

**Reply by ShafSpecs** [reply]
https://remix-pwa.run/docs/upgrade-guide

it can be migrated to. You can still wait a short while before upgrading if you want too

**Reply by rista404** [reply]
@ShafSpecs seems like there's no v3 on npm? The latest version of `@remix-pwa/dev` for example is [2.0.22](https://www.npmjs.com/package/@remix-pwa/dev/v/2.0.22)

**Reply by ShafSpecs** [reply]
https://remix-pwa.run/docs/versioning

That's right, versioning works a bit weird in remix-pwa. I explained why in this short doc
